pymeasure/instruments/keithley/keithley2602B.py

```diff
@@ -0,0 +1,212 @@
+
+    def write(self, cmd):
+        self.instrument.write(f"digio.trigger.[{self.pin_number}].{cmd}")
+
+    def check_errors(self):
+        return self.instrument.check_errors()
+
+    def assert_trigger(self):
+        """This method asserts a trigger pulse on one of the digital I/O lines."""
+
+        log.info(f"Asserting a trigger pulse on pin number {self.pin_number}.")
+        self.write("assert()")
+        self.check_errors()
+
+    def clear_trigger(self):
+        """This method clears the trigger event detector on a digital I/O line."""
+
+        log.info(f"Clearing trigger on pin number {self.pin_number}.")
+        self.write("clear()")
+        self.check_errors()
+
+    def get_event_id(self):
+        """This method returns the mode in which the trigger event detector and
+        the output trigger generator operate on the given trigger line. See description
+        of all the possible EVENT_IDs on page 9-57 of the Series 2600B Reference Manual.
+        """
+
+        id = self.ask("EVENT_ID")
+        self.check_errors()
+        return int(id)
+
+    def get_overrun_status(self):
+        """This method returns the event detector overrun status. If this is
+        true, an event was ignored because the event detector was already in the
+        detected state when the event occurred. This is an indication of the
+        state of the event detector built into the line itself. It does not
+        indicate if an overrun occurred in any other part of the trigger model
+        or in any other detector that is monitoring the event."""
+
+        response_status = self.ask("overrun")
+        self.check_errors()
+
+        if response_status == "false":
+            status = False
+        elif response_status == "true":
+            status = True
+        else:
+            status = None
+
+        return status
+
+    def release_trigger(self):
+        """This method releases an indefinite length or latched trigger."""
+
+        log.info(f"Releasing trigger on pin number {self.pin_number}.")
+        self.write("release()")
+        self.check_errors()
+
+    def reset_trigger_values(self):
+        """This method resets trigger values to their factory defaults. It
+        sets the mode, pulsewidth, stimulus, and overrun status to factory
+        default settings.
+        """
+
+        log.info(
+            f"Resetting trigger values (to factory defaults) on pin number {self.pin_number}."
+        )
+        self.write("reset()")
+        self.check_errors()
+
+    def wait_for_trigger(self, timeout):
+        """This method waits for a trigger for up to a maximum of the timeout value (in seconds).
+        Returns True if a trigger was detected, false if the timout was reached and no trigger was
+        detected.
+        """
+
+        log.info(f"Waiting for trigger for {timeout} on pin number {self.pin_number}.")
+        self.write("wait(timeout)")
+        self.check_errors()
```

Comment:
Do these make sense as methods, or should they be `Instrument.control` properties?

Reply:
I reckon properties to be appropriate getting or setting values, properties etc. So get_event_id and get_oberrun could be properties (measurement, I guess).

Waiting on a trigger or clearing etc are, however, methods which I execute on the device, therefore, it's appropriate to leave them methods. At least in my understanding.

I'd say as a rule of thumb: If you call it get_x or set_x, it should be a property. If your method neither takes a parameter nor returns something, it should be a method.